lib/irb/ruby-lex.rb

```diff
@@ -237,11 +188,12 @@ def readmultiline
       # Accept any single-line input for symbol aliases or commands that transform args
       return code if single_line_command?(code)
 
-      ltype, indent, continue, code_block_open = check_code_state(code)
-      return code unless ltype or indent > 0 or continue or code_block_open
+      check_target_code, tokens, opens = check_code_state(code)
+      continue = process_continue(tokens)
```

Comment:
I know continue is a boolean state but it's not very clear what it means. Would you mind explain it a bit?
Also, I feel `is_continue?` or `should_continue?` would probably be a much better name. Do you mind opening a PR to rename it first? It should help others understand this PR a bit easier.

Reply:
`!process_continue(...) && !check_code_block(...)` is checking if the input still continues or not.

`process_continue` it is checking if the line ends with `\`, means user wants to write next line, and doing more thing.
`check_code_block` is checking if syntax-error code is recoverable or not, and doing more thing.

I actually don't know what it is doing on each method. I think there's overlap or duplication. IMO, later part of check_code_block where last_lex_state is used looks like it should belong to process_continue.

Reply:
Thanks for the explanation. And yeah refactoring it later seems better.